### PR TITLE
Misc for lib

### PR DESCRIPTION
**Pull request by privat**
Litttle features and warning-fixes for the libs

### CODE REVIEW

**Comment by R4PaSs** [comment]
:shipit:

File: lib/standard/collection/abstract_collection.nit [hunk 1]
```diff
@@ -66,6 +66,14 @@ interface Collection[E]
 	#     assert [1..1[.is_empty   == true
 	fun is_empty: Bool do return length == 0
 
+	# Alias for `not is_empty`.
```
Comment (Morriar) [inline lib/standard/collection/abstract_collection.nit]: Do we need to duplicate all boolean services from now?

I'm not a big fan of that one...

File: lib/standard/collection/sorter.nit [hunk 2]
```diff
@@ -250,6 +250,81 @@ interface Comparator
...
+	# See: `to_map_comparator` to get the comparator used internally.
+	fun keys_sorted_by_values(comparator: Comparator): Array[K]
+	do
+		var keys = self.keys.to_a
+		var map_cmp = to_map_comparator(comparator)
+		map_cmp.sort(keys)
+		return keys
+	end
+
+	# A comparator that compare things with their values in self.
```
Comment (Morriar) [inline lib/standard/collection/sorter.nit]: compares*

File: lib/standard/collection/sorter.nit [hunk 3]
```diff
@@ -250,6 +250,81 @@ interface Comparator
...
+	#
+	# See `MapComparator` for details.
+	fun to_map_comparator(comparator: Comparator): MapComparator[K, V] do return new MapComparator[K,V](self, comparator)
+end
+
+# A comparator that compare things with their values in a map
```
Comment (Morriar) [inline lib/standard/collection/sorter.nit]: compare*

File: lib/standard/collection/sorter.nit [hunk 4]
```diff
@@ -250,6 +250,81 @@ interface Comparator
...
+# map_cmp.sort(a)
+# assert a == ["one", "ten", "two"]
+# ~~~
+class MapComparator[K,V]
+	super Comparator
+
+	# What is compared are the keys of the values
+	redef type COMPARED: K
+
+	# The map that associate compared elements to the value used to compare them
```
Comment (Morriar) [inline lib/standard/collection/sorter.nit]: associates*

**Comment by Morriar** [comment]
+1

**Comment by BlackMinou** [comment]
:shipit:

File: lib/standard/collection/abstract_collection.nit (hunk 1 above)
Comment (xymus) [inline lib/standard/collection/abstract_collection.nit]: It's worth it! I use more `not ...is_empty` than the positive. A quick git grep in WBTW revealed that it uses the positive only 24 times, vs 44 times for the `not ...is empty`.

**Comment by xymus** [comment]
:shipit:

File: lib/standard/collection/abstract_collection.nit (hunk 1 above)
Comment (privat) [inline lib/standard/collection/abstract_collection.nit]: another way could be to add a generalized boolean converter. `to_b`? `is_good`?

**Pull request merged**